src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -957,7 +957,17 @@ QString QgsOgrProviderUtils::connectionPoolId( const QString &dataSourceURI, boo
     QString filePath = dataSourceURI.left( dataSourceURI.indexOf( QLatin1Char( '|' ) ) );
     QFileInfo fi( filePath );
     if ( fi.isFile() )
-      return filePath;
+    {
+      // Preserve open options so pooled connections always carry those on
+      QString openOptions;
+      static thread_local QRegularExpression openOptionsRegex( QStringLiteral( "((?:\\|option:(?:[^|]*))+)" ) );
```

Comment:
couldn't we use QgsOgrProviderMetadata::decodeUri() instead ?

Reply:
@rouault , we could, but I was worried it'd be a bit too costly for no real reason (we're not re-encoding afterwards, we're not in need of other bits either).